File: apps/mobile/lib/features/community/domain/models/sponsorship_details.dart
import 'package:meta/meta.dart';

@immutable
class SponsorshipDetails {
  final String goalDescription;
  final double? estimatedAmountLkr;
  final String? adminRejectionNotes;
  final DateTime? fundedAt;
  final int proofDocumentCount;

  const SponsorshipDetails({
    required this.goalDescription,
    this.estimatedAmountLkr,
    this.adminRejectionNotes,
    this.fundedAt,
    required this.proofDocumentCount,
  });
}

File: apps/api/HappyPaws.Domain/Entities/SponsorshipDetails.cs
using System;

namespace HappyPaws.Domain.Entities;

public class SponsorshipDetails
{
    public Guid PostId { get; set; }
    public string GoalDescription { get; set; } = null!;
    public decimal? EstimatedAmountLkr { get; set; }
    public long? AdminApproverId { get; set; }
    public DateTimeOffset? AdminReviewedAt { get; set; }
    public string? AdminRejectionNotes { get; set; }
    public DateTimeOffset? FundedAt { get; set; }

    public Post Post { get; set; } = null!;
    public User? AdminApprover { get; set; }
}
